nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/LocalRequestTraceStore.java

```diff
@@ -98,20 +93,20 @@ public RequestTrace addTrace(RequestTrace trace, RequestTrace traceToRemove) {
 
     @Override
     public void setSize(int maxSize) {
-        // Set the store to be one bigger than the specified value. Allows the store to transparently select which item to remove
-        this.store = new BoundedTreeSet<>(maxSize + 1);
+        while (store.size() > maxSize) {
+            store.remove(strategy.getTraceForRemoval(getTraces(), maxSize, null));
+        }
         this.maxStoreSize = maxSize;
     }
 
     @Override
     public int getStoreSize() {
-        return this.store.size();
+        return maxStoreSize;
```

Comment:
Its unclear what `getStoreSize()` should return since one strategy would return the `maxSize` while this was returning the actual number of values. I felt that it was meant to return the `maxSize` as the value size could be obtained by `getTraces().size()` while there was no other way to obtain the `maxSize`.